react/testing.md

```diff
@@ -1,3 +1,291 @@
+* @param {ReactWrapper} componentWrapper - Rendered componentWrapper (result of mount, shallow, or render)
+* @param {string} specName - Name of `data-spec` attribute value to find
+* @param {string|Function} typeFilter - (Optional) Expected type of the wrappers (defaults to all HTML tags)
+* @returns {ReactComponent} Single matching DOM component
+*/
+export const getSingleSpecWrapper = (componentWrapper, specName, typeFilter) => {
+
+    let specWrappers = getSpecWrappers(componentWrapper, specName, typeFilter);
+
+    if (specWrappers.length !== 1) {
+        throw new Error(`Expected single "${specName}" spec wrapper. Received: ${specWrappers.length}.`);
+    }
+
+    return specWrappers.first();
+};
+```
+
+**[⬆ back to top](#table-of-contents)**
+
+## Finding components
+
+You can find a component simply by using Enzyme's [`find`](http://airbnb.io/enzyme/docs/api/ReactWrapper/find.html) and passing the component class:
+
+```js
+it('should render a checked checkbox if it is selected', () => {
+    let wrapperInstance = mount(<Component isSelected={true} />);
+    let checkboxWrapper = wrapperInstance.find(Checkbox);
+
+    expect(checkboxWrapper).to.have.prop('isChecked', true);
+});
+```
+
+This works as long as there's only one `Checkbox` rendered within `Component`. If there are multiple `Checkbox` components within `Component`, `checkboxWrapper` would have multiple elements in it. Instead you can add a `data-spec` attribute to the specific `Checkbox` and use `getSingleSpecWrapper`:
+
+```js
+// good
+it('should render a checked checkbox if it is selected', () => {
+    let wrapperInstance = mount(<Component isSelected={true} />);
+
+    // pass the component class as the third parameter to `getSingleSpecWrapper`
+    let selectAllCheckboxWrapper = getSingleSpecWrapper(wrapperInstance, 'component-selectAll', Checkbox);
+
+    expect(selectAllCheckboxWrapper).to.have.prop('isChecked', true);
+});
+
+// bad (finds the appropriate Checkbox based on source order)
+it('should render a checked checkbox if it is selected', () => {
+    let wrapperInstance = mount(<Component isSelected={true} />);
+    let selectAllCheckboxWrapper = wrapperInstance.find(Checkbox).at(2);
+
+    expect(selectAllCheckboxWrapper).to.have.prop('isChecked', true);
+});
+```
+
+The key in the "good" example is the third parameter passed to `getSingleSpecWrapper`. By default `getSingleSpecWrapper` will try to find a node with the specified `data-spec`. But if you specify the component class (`Checkbox` in this case), it'll return a reference to the component wrapper.
+
+**[⬆ back to top](#table-of-contents)**
+
+## Testing existence
+
+### Testing node existence
+
+To [find nodes](#finding-nodes) you use the `getSingleSpecWrapper` & `getSpecWrappers` helpers.
+
+`getSingleSpecWrapper` throws an error if a single node is not found. Therefore, in order to determine if a node exists you actually to test whether or not calling the helper throws an `Error`:
```

Comment:
Remove `to`

Should be:

+`getSingleSpecWrapper` throws an error if a single node is not found. Therefore, in order to determine if a node exists you actually test whether or not calling the helper throws an `Error`: